### DISCUSSION TITLE
How do I set roughness in the viewer?

### DISCUSSION THREAD

**Discussion by TruPhnxKng** [google/model-viewer#4033] [Q&A]
How do I set roughness in the viewer?  I'm uploading cell shaded models and they are so shinny they look like porcelain. 

I just want to know the line of code needed to make the texture rougher.

**Comment by elalish** [comment]
Ideally you set the roughness when you export the glTF - the default is 1 (fully rough), so if yours are shiny something is intentionally setting it to a different value. You can fix it in our [editor](https://modelviewer.dev/editor/), or if you really need to you can use our material API: `mv.model.materials[0].pbrMetallicRoughness.setRoughnessFactor(1);` https://modelviewer.dev/examples/scenegraph/#pickMaterialExample
